### DISCUSSION TITLE
Artificial Diversity Correction

### DISCUSSION THREAD

**Discussion by bshim181** [milaboratory/mixcr#1609] [Q&A] [answered]
Hello, 

I am running generic-amplicon preset for my TCR data with UMI length of 12. I am getting high percentage of aritifical diversity eliminated and was wondering why this might be. 

<img width="504" alt="Screenshot 2024-04-02 at 5 04 05 PM" src="https://github.com/milaboratory/mixcr/assets/53489568/4e7cfd04-0ddf-437d-b49c-33bf04778edd">

I am wondering if I specified the parameters wrong. This is what my read architecture resembles. 

<img width="493" alt="Screenshot 2024-04-02 at 5 11 43 PM" src="https://github.com/milaboratory/mixcr/assets/53489568/16c6b6e7-ebd5-4e6d-a610-831a45e360d0"> 

This is the parameter that I executed. 

  tag_pattern='^(UMI:N{12})(R1:*)\^(R2:*)'
  java -Xmx60g -jar $mixcr analyze generic-amplicon-with-umi \
      --species hsa \
      --assemble-clonotypes-by CDR3 \
      --export-productive-clones-only \
      --rna \
      -f \
      --tag-pattern $tag_pattern \
      -Massemble.cloneAssemblerParameters.addReadsCountOnClustering=true \
      --rigid-left-alignment-boundary \
      --floating-right-alignment-boundary C \
      ${R1_file} \
      ${R2_file} \
      ${out_dir}/${filename_woExt}_${run}/${filename_woExt}

**Comment by mizraelson** [comment, marked as answer]
Hi, the results  look fine, the important thing here is that the `records weight accepted` is 98.68%, meaning most of the reads were ppreserved.
